### PR TITLE
 ops: kokkos: revise rank-1 update

### PR DESCRIPTION
**Pull request by mzuzek**
Contents:
- [x] revised overload constraints 
- [x] added extent checks
- [x] added explicit type casting for coefficients
- [x] fixed NaN injection bug added related unit tests
- [x] added expression unit tests with `span` and `diag`

refs #448

### CODE REVIEW

File: tests/functional_small/ops/ops_kokkos_vector.cc [hunk 1]
```diff
@@ -306,6 +306,87 @@ TEST(ops_kokkos, vector_update4)
   EXPECT_DOUBLE_EQ( v_h2(2), 12.0);
 }
 
+TEST(ops_kokkos, vector_update_nan1)
+{
+  Kokkos::View<double*> v("v", 3);
+  Kokkos::View<double*> a("a", 3);
+  Kokkos::View<double*> nan("nan", 3);
```
Comment (fnrizzi) [inline tests/functional_small/ops/ops_kokkos_vector.cc:313]: rename `nan -> vecOfNan`

**Comment by fnrizzi** [comment]
can you please also fix the typo at the top:
```
struct _kokkosUpdateAdmissibleOperands
{
  /* make sure we don't pass const objects to be modified.
     In kokkos it is legal to modify const views, not for pressio wrappers. */
  static_assert
    (!std::is_const<T1>::value,
     "ops:product: cannot modify a const-qualified wrapper of a Kokkos view");
```     
ops::product -> ops::update

File: tests/functional_small/ops/ops_kokkos_vector.cc (hunk 1 above)
Comment (mzuzek) [reply tests/functional_small/ops/ops_kokkos_vector.cc]: ok, fixed on 8c932da1af8c8bc66afbfad11f0ee5147380076b

**Comment by mzuzek** [comment]
Thanks @fnrizzi for your comments!
I've added expression tests on 1242f193033a5e5a0c894e60bc8e9d1095b1ec44.

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**